packages/components/src/components/teaser/teaser.stories.ts

```diff
@@ -0,0 +1,352 @@
+const { overrideArgs } = storybookHelpers('sd-teaser');
+const { generateTemplate } = storybookTemplate('sd-teaser');
+
+export default {
+  title: 'Components/sd-teaser',
+  component: 'sd-teaser',
+  args: overrideArgs([
+    {
+      type: 'slot',
+      name: 'default',
+      value: `<slot-comp style="--slot-height: 48px; --slot-content: 'Main slot'"></slot-comp>`
+    },
+    {
+      type: 'slot',
+      name: 'media',
+      value: `<slot-comp slot='media' style="--slot-height: 48px; --slot-content: 'Media slot'"></slot-comp>`
+    },
+    {
+      type: 'slot',
+      name: 'headline',
+      value: '<h2 slot="headline">Remember: Always Use &lt;h&gt; Tags here!</h2>'
+    },
+    {
+      type: 'slot',
+      name: 'meta',
+      value: `<slot-comp slot='meta' style="--slot-height: 46px; --slot-content: 'Meta slot'"></slot-comp>`
+    }
+  ]),
+  argTypes,
+  parameters: { ...parameters }
+};
+
+/**
+ * This shows sd-teaser in its default state.
+ */
+
+export const Default = {
+  render: (args: any) => {
+    return generateTemplate({ args });
+  }
+};
+
+/**
+ * Teaser in all possible variants.
+ */
+
+export const Variants = {
+  parameters: { controls: { exclude: 'variant' } },
+  render: (args: any) => {
+    return generateTemplate({
+      axis: {
+        y: { type: 'attribute', name: 'variant' }
+      },
+      args,
+      constants: [
+        {
+          type: 'template',
+          name: 'style',
+          value: '<div style="margin-bottom: 50px">%TEMPLATE%</div>'
+        },
+        {
+          type: 'attribute',
+          name: 'orientation',
+          value: 'vertical'
+        }
+      ]
+    });
+  }
+};
+
+/**
+ * Teaser in all possible combinations of `inset` and 'orientation'. Note that in case the variant is `white border-neutral-300`, the inset is always `true`.
+ */
+
+export const InsetAndOrientation = {
+  name: 'Inset x Orientation',
+  parameters: { controls: { exclude: ['inset', 'orientation'] } },
+  render: (args: any) => {
+    return generateTemplate({
+      axis: {
+        x: { type: 'attribute', name: 'inset' },
+        y: { type: 'attribute', name: 'orientation' }
+      },
+      args,
+      constants: [
+        {
+          type: 'attribute',
+          name: 'variant',
+          value: 'primary-100'
+        }
+      ]
+    });
+  }
+};
+
+/**
+ * Teaser with different media and content distribution values.
+ */
+
+export const DistributionRatio = {
+  parameters: { controls: { exclude: ['--distribution-media', '--distribution-content'] } },
+  render: (args: any) => {
+    return generateTemplate({
+      axis: {
+        y: {
+          type: 'template',
+          name: 'media and content distribution',
+          values: [
+            {
+              title: 'media: 66%, content: 33%',
+              value: '<div style="--distribution-media: 66%; --distribution-content: 33%;">%TEMPLATE%</div>'
+            },
+            {
+              title: 'media: 200px, content: 450px',
```

Comment:
would content 100% work? I think it's more realistic to have only one fixed value and let the other one resize automatically.

Reply:
No, this is totally fine like this.